presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchQueryBuilder.java

```diff
@@ -39,37 +41,42 @@
 import static io.prestosql.spi.type.BooleanType.BOOLEAN;
 import static io.prestosql.spi.type.DoubleType.DOUBLE;
 import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.SmallintType.SMALLINT;
+import static io.prestosql.spi.type.TimestampType.TIMESTAMP;
+import static io.prestosql.spi.type.TinyintType.TINYINT;
 import static io.prestosql.spi.type.VarcharType.VARCHAR;
+import static java.lang.Math.toIntExact;
+import static java.time.format.DateTimeFormatter.ISO_DATE_TIME;
 
-public class ElasticsearchQueryBuilder
+public final class ElasticsearchQueryBuilder
 {
     private ElasticsearchQueryBuilder() {}
 
-    public static QueryBuilder buildSearchQuery(TupleDomain<ColumnHandle> constraint, List<ElasticsearchColumnHandle> columns, Optional<String> query)
+    public static QueryBuilder buildSearchQuery(ConnectorSession session, TupleDomain<ElasticsearchColumnHandle> constraint, Optional<String> query)
     {
-        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
-        for (ElasticsearchColumnHandle column : columns) {
-            BoolQueryBuilder columnQueryBuilder = new BoolQueryBuilder();
-            Type type = column.getType();
-            if (constraint.getDomains().isPresent()) {
-                Domain domain = constraint.getDomains().get().get(column);
-                if (domain != null) {
-                    columnQueryBuilder.should(buildPredicate(column.getName(), domain, type));
+        BoolQueryBuilder queryBuilder = new BoolQueryBuilder();
+        if (constraint.getDomains().isPresent()) {
+            for (Map.Entry<ElasticsearchColumnHandle, Domain> entry : constraint.getDomains().get().entrySet()) {
+                ElasticsearchColumnHandle column = entry.getKey();
+                Domain domain = entry.getValue();
+
+                checkArgument(!domain.isNone(), "Unexpected NONE domain for %s", column.getName());
+                if (!domain.isAll()) {
+                    queryBuilder.must(new BoolQueryBuilder().must(buildPredicate(session, column.getName(), domain, column.getType())));
```

Comment:
to enable query caching on ES and also to avoid scoring (which is irrelevant anyway since you are doing a scroll) this should really be `filter` and not a `must`

Reply:
Thanks for the pointers! I'll put up an update with those changes.

Reply:
https://github.com/prestosql/presto/pull/2402